src/efftester.ml

```diff
@@ -99,14 +103,25 @@ type lit =
   | LitStr of string
   | LitList of lit list
 
+(** type term is used to represent all syntax constructs (of OCaml) available in our
+generator  *)
 type term =
   | Lit of lit
   | Variable of etype * variable
+  (* [Constructor (type, name, load)] is used to construct ADT variants *)
+  | Constructor of etype * string * term option
+  (* [PatternMatch typ matched_trm branches eff]
+    Note: in type (term * term) list, the first part of the tuple must be a value (no evaluation)
+  *)
+  | PatternMatch of etype * term * (term * term) list * eff
   | Lambda of etype * variable * etype * term
   | App of etype * term * etype * term * eff
   | Let of variable * etype * term * term * etype * eff
   | If of etype * term * term * term * eff
 
+let some load_typ load = Constructor (Option load_typ, "Some", Some load)
```

Comment:
Out of curiosity, why the `load_` in `load_typ`?

Reply:
So that one would not be puzzled (even for a sec) if s/he needs to provide the type of the load `t` or `Option (t)` while using function `some`.

Reply:
I think "payload" is the more common term (I find "load" a bit confusing), but I would use "argument" instead of "payload" for constructors.